homeassistant/components/sensor/pollen.py

```diff
@@ -192,187 +169,162 @@ def state(self):
-            self._attrs[ATTR_SEASON] = None
-
-        try:
-            self._attrs[ATTR_TREND] = self.data.outlook_data['Trend'].title()
-        except KeyError:
-            _LOGGER.debug('Trend data not included in API response')
-            self._attrs[ATTR_TREND] = None
-
-        self._state = period['Index']
-        self._unit = 'index'
-
-
-class DataBase(object):
-    """Define a generic data object."""
-
-    def __init__(self, client):
+            for idx, attrs in enumerate(period['Triggers']):
+                index = idx + 1
+                self._attrs.update({
+                    '{0}_{1}'.format(ATTR_ALLERGEN_GENUS, index):
+                        attrs['Genus'],
+                    '{0}_{1}'.format(ATTR_ALLERGEN_NAME, index):
+                        attrs['Name'],
+                    '{0}_{1}'.format(ATTR_ALLERGEN_TYPE, index):
+                        attrs['PlantType'],
+                })
+
+            self._attrs.update({
+                ATTR_CITY: data['City'].title(),
+                ATTR_RATING: rating,
+                ATTR_STATE: data['State'],
+                ATTR_ZIP_CODE: data['ZIP']
+            })
+            self._state = period['Index']
+        elif self._type == TYPE_DISEASE_FORECAST:
+            self._attrs.update({
+                ATTR_CITY: data['City'].title(),
+                ATTR_RATING: rating,
+                ATTR_STATE: data['State'],
+                ATTR_TREND: trend,
+                ATTR_ZIP_CODE: data['ZIP']
+            })
+            self._state = average
+
+
+class PollenComData(object):
+    """Define a data object to retrieve info from Pollen.com."""
+
+    def __init__(self, client, sensor_types):
         """Initialize."""
         self._client = client
+        self._sensor_types = sensor_types
+        self.data = {}
 
-    def _get_client_data(self, module, operation):
-        """Get data from a particular point in the API."""
-        from pypollencom.exceptions import HTTPError
-
-        data = {}
-        try:
-            data = getattr(getattr(self._client, module), operation)()
-            _LOGGER.debug('Received "%s_%s" data: %s', module, operation, data)
-        except HTTPError as exc:
-            _LOGGER.error('An error occurred while retrieving data')
-            _LOGGER.debug(exc)
-
-        return data
-
-
-class AllergyAveragesData(DataBase):
-    """Define an object to averages on future and historical allergy data."""
-
-    def __init__(self, client):
-        """Initialize."""
-        super().__init__(client)
-        self.extended_data = None
-        self.historic_data = None
-
-    @Throttle(MIN_TIME_UPDATE_AVERAGES)
-    def update(self):
-        """Update with new data."""
-        self.extended_data = self._get_client_data('allergens', 'extended')
-        self.historic_data = self._get_client_data('allergens', 'historic')
-
+        self.async_update = Throttle(DEFAULT_SCAN_INTERVAL)(self._async_update)
```

Comment:
Now you can apply the throttle as a decorator which is standard and define the method directly as `async_update`.

Reply:
Aha! Yes!